src/Cronofy/AddToCalendarRequestBuilder.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace Cronofy
+{
+    using System;
+    using Cronofy.Requests;
+
+    /// <summary>
+    /// Class to build an add to calendar request.
+    /// </summary>
+    public sealed class AddToCalendarRequestBuilder : IBuilder<AddToCalendarRequest>
+    {
+        /// <summary>
+        /// The oauth details for the request.
+        /// </summary>
+        private AddToCalendarRequest.OAuthDetails oauth;
+
+        /// <summary>
+        /// The event details builder for the request.
+        /// </summary>
+        private UpsertEventRequestBuilder upsertEventRequestBuilder;
+
+        /// <summary>
+        /// The event details for the request.
+        /// </summary>
+        private UpsertEventRequest upsertEventRequest;
+
+        /// <summary>
+        /// Sets the OAuth details of the request.
+        /// </summary>
+        /// <param name="redirectUrl">
+        /// The redirect url for the request's oauth details, must not be blank.
+        /// </param>
+        /// <param name="scope">
+        /// The scope for the request's oauth details, must not be blank.
+        /// </param>
+        /// <param name="state">
+        /// The state for the request's oauth details.
+        /// </param>
+        /// <returns>
+        /// A reference to the <see cref="AddToCalendarRequestBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Thrown if <paramref name="redirectUrl"/> or <paramref name="scope"/> are empty.  
+        /// </exception>
+        public AddToCalendarRequestBuilder OAuthDetails(string redirectUrl, string scope, string state = null)
```

Comment:
"URL" rather than "URI" is going to filter through to here too. Not sure if optional arguments are in the lowest version of .NET we are targeting or not.

We should also create a "builder" overload so there's an option that doesn't involve 3 string arguments in a row.